### PR TITLE
windows: add VT100 codes ?25l and ?25h (hide/show cursor)

### PR DESCRIPTION
**Pull request by Qix-**
One of the first machines to support ANSI was the VT100; two of its extension codes, `?25l` and `?25h` (hide/show cursor, respectively) are still in use today (and are supported by most/all ANSI-compliant terminals).

This patch adds support for this in Windows using `Get`/`SetConsoleCursorInfo` and introduces a skipped character `?` to the ANSI parser.

> An initial `?` is innocuous to _any_ escape sequence; this means this will not break if `CSI [?31m` is outputted; as expected, dark red text is shown.

### CODE REVIEW

**Comment by Qix-** [comment]
> The following commiters were not found in the CLA

I'm in the CLA. [I promise](https://github.com/joyent/libuv/pull/1243#issuecomment-40914261). :)

> First line of commit message must be no longer than 50 characters

Is this fixed by a `commit --amend` / `push --force`?

File: src/win/tty.c [hunk 1]
```diff
@@ -1527,6 +1545,13 @@ static int uv_tty_write_bufs(uv_tty_t* handle,
               continue;
             }
 
+          } else if (utf8_codepoint == '?' && !(ansi_parser_state & ANSI_IN_ARG) &&
+                     handle->ansi_csi_argc == 0) {
+            /* Ignores '?' if it is the first character after CSI[ */
```
Comment (saghul) [inline src/win/tty.c:1550]: Comment the whole block, not each line.

**Comment by saghul** [comment]
> Is this fixed by a commit --amend / push --force?

Yes.

Code LGTM, but I have no experience in terminal handling, so I'll wait for @indutny or @piscisaureus to comment as well.

**Comment by piscisaureus** [comment]
Commits to be squashed, but the code is otherwise flawless. Thanks.

File: src/win/tty.c (hunk 1 above)
Comment (Qix-) [inline src/win/tty.c:1550]: Check 1517/1535 - That seems to be the commenting scheme.

**Comment by Qix-** [comment]
@piscisaureus Fixed the commit message. :) Thanks for the review.

**Comment by saghul** [comment]
@Qix- rebased and landed in https://github.com/joyent/libuv/commit/21b1b87d12a36e7912a8237241267402c91e5162 Thanks a lot for this and the other pending PRs, keep 'em coming! ;-)

**Pull request closed without merging**